scp.c

```diff
@@ -294,6 +294,10 @@ do_cmd(char *host, char *remuser, int port, char *cmd, int *fdin, int *fdout)
 #ifdef WINDOWS
 	/* generate command line and spawn_child */
 	replacearg(&args, 0, "%s", ssh_program);
+	if (port != -1) {
```

Comment:
the same change need to added in function do_cmd2, line 380

Reply:
fixed